src/main/java/org/jabref/logic/importer/RawFetcher.java

```diff
@@ -0,0 +1,56 @@
+package org.jabref.logic.importer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.util.Collections;
+import java.util.List;
+
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.strings.StringUtil;
+
+public interface RawFetcher extends SearchBasedParserFetcher {
+
+    /**
+     * Send queries with arbitrary url parameters.
+     * <p>
+     * This is used to test the library APIs for their advanced search options e.g. fielded search.
+     *
+     * @param urlParameters The given URL parameters as string should start with "?"
+     * @return result of the query with the given URL parameters
+     */
+    default List<BibEntry> performRawSearch(String urlParameters) throws FetcherException, IllegalArgumentException {
+        if (StringUtil.isBlank(urlParameters)) {
+            return Collections.emptyList();
+        }
+        if (urlParameters.startsWith("?")) {
+            urlParameters = urlParameters.replaceFirst("\\?", "");
+        }
+
+        // Replace white spaces with +
```

Comment:
```suggestion
        // Replace white spaces by + to form a valid URL parameter
```

(Why don't we need a full URL encoding?)

Reply:
If we use full URL encoding, it would also encode the "&" sign not allowing us to add any URL parameters.